libsupport/include/katana/IndexID.h

```diff
@@ -0,0 +1,75 @@
+#ifndef KATANA_LIBSUPPORT_KATANA_INDEXID_H_
+#define KATANA_LIBSUPPORT_KATANA_INDEXID_H_
+
+#include "katana/OpaqueID.h"
+
+namespace katana {
+
+/// A numeric ID type suitable for indexing. IndexIDs support:
+///
+/// - Addition and subtraction of other IDs (the result of which is an ID)
+/// - Increment and decrement
+/// - Dereference
```

Comment:
Please add a note here that we should prefer `OpaqueID` and `OpaqueIDLinear` to this because those provide more safety. This type will allow accidental math with int which was part of what opaque IDs were designed to prevent. So while there will be cases where this is appropriate people should use the stricter variants if possible.